server/routes/index.ts

```diff
@@ -16,14 +15,36 @@ router.get('/health', async (req, res) => {
   res.send('OK').end()
 })
 
+const substackCache: Record<string, { cacheTime: number, headers: unknown, data: unknown }> = {}
+const cacheLife = 1000 * 60
+
 router.get('/substack/api/v1/:endpoint',
   limiter(),
   substack,
   async (req, res) => {
     const { substackDomain } = res.locals
     const { endpoint } = req.params
     try {
-      const { headers, data } = await axiosBase.get(`https://${substackDomain}/api/v1/${endpoint}`, { params: req.query })
+      const url = `https://${substackDomain}/api/v1/${endpoint}`
+      const cacheKey = `${url}-${JSON.stringify(req.query)}`
+
+      let headers, data
+
+      if (substackCache[cacheKey]?.cacheTime !== undefined && substackCache[cacheKey].cacheTime + cacheLife > Date.now()) {
```

Comment:
use LRUCache to manage life, it has multiple modes such as auto-refetch but report as hit instead of miss, which is pretty useful for production